### PR TITLE
Issue #187 - bug - gatekeeper expander - wrong event_type set for generated inform-gatekeeper-admission-* ConfigurationPolicy

### PR DESCRIPTION
**Pull request by itewk**
Resolves issue #187

### CODE REVIEW

**Review by mprahl — APPROVED** [review]

**Review by dhaiducek — CHANGES_REQUESTED** [review]
@itewk It's not actually incorrect. The event_type for admission events is indeed `violation`:

https://github.com/open-policy-agent/gatekeeper/blob/0a4851a975d25003218a23b8a70217f759f2454f/pkg/webhook/policy.go#L297

However, we'd welcome a contribution to add the `violation_audit` to the expander.

**Review by mprahl — CHANGES_REQUESTED** [review]
Thanks for catching that @dhaiducek!

File: internal/expanders/gatekeeper.go [hunk 1]
```diff
@@ -110,7 +110,7 @@ func (g GatekeeperPolicyExpander) Expand(
 								"constraint_action": "deny",
 								"constraint_kind":   constraintKind,
 								"constraint_name":   constraintName,
-								"event_type":        "violation",
+								"event_type":        "violation_audited",
 							},
 						},
 					},
```
Comment (dhaiducek) [inline internal/expanders/gatekeeper.go:116]: The simplest thing might be to just add it as another `object-template`. What would you think of that, @itewk?
```suggestion
					},
					{
						"complianceType": "mustnothave",
						"objectDefinition": map[string]interface{}{
							"apiVersion": "v1",
							"kind":       "Event",
							"annotations": map[string]interface{}{
								"constraint_action": "deny",
								"constraint_kind":   constraintKind,
								"constraint_name":   constraintName,
								"event_type":        "violation",
								"event_type":        "violation_audited",
							},
						},
					},
```

**Comment by itewk** [comment]
aaah. okay.

File: internal/expanders/gatekeeper.go (hunk 1 above)
Comment (itewk) [reply internal/expanders/gatekeeper.go:116]: yeah. but im also fighting with the hard coded namespaceSelector and the note left behind about "futer improvements needed". its annoying having namepsaces /excludedNamespaces set in my constraint (both at the gatekeeper and OPM level) but then those don't propigate to the generated ConfigurationPolicy(s) which ar harded cluded eclude/includes.

i am going to keep playing with it to see if i can get the behavor i would expect.

**Pull request closed without merging**